dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/DdAndroidGradlePlugin.kt

```diff
@@ -101,6 +104,69 @@ class DdAndroidGradlePlugin : Plugin<Project> {
         return uploadTask
     }
 
+    @Suppress("DefaultLocale")
+    internal fun configureVariantForSdkCheck(
+        target: Project,
+        variant: ApplicationVariant,
+        extension: DdExtension
+    ): Task? {
+
+        if (!extension.enabled) {
+            return null
+        }
+
+        // variants will have name like proDebug, but compile task will have a name like
+        // compileProDebugSources. It can be other tasks like compileProDebugAndroidTestSources
+        // or compileProDebugUnitTestSources, but we are not interested in these. This is fragile
+        // and depends on the AGP naming convention
+        val compileTask = target.tasks
+            .findByName("compile${variant.name.capitalize()}Sources")
+
+        if (compileTask == null) {
+            LOGGER.warn(
```

Comment:
Maybe we could use the same level of warning for this step as for the actual check, WDYT?

Reply:
I wouldn't link it to the actual check, actually this branch exists because `findByName("compile${variant.name.capitalize()}Sources")` above is kind of fragile - it relies on the naming convention introduced by the current version of AGP, which may change in the future versions. So I was thinking about receiving feedback from the users who use some `alpha/beta` versions of AGP where this convention _may be_ different. In theory this "should never happen" ©️ , because then AGP will break backward compatibility with a bunch of other plugins I think. Do you think users will miss this warning?